projects/components/src/form/base-form-control.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Input } from '@angular/core';
+import { ControlValueAccessor, FormControl, NgControl } from '@angular/forms';
+import { IdGenerator } from '../utils/id-generator/id-generator';
+import { CanBeReadOnly } from './interfaces/can-be-read-only.interface';
+
+const idGenerator = new IdGenerator('base-form-control-id');
+
+export abstract class BaseFormControl implements ControlValueAccessor, CanBeReadOnly {
```

Comment:
Please create good documentation for this. Explaining that this is a wrapper to enforce our form UX decisions like readonly-ness, label position and error displaying.